### ISSUE TITLE
SA1629 should not ignore enum members

### ISSUE THREAD

**Issue by RalfKoban** [DotNetAnalyzers/StyleCopAnalyzers#2950]
_(StyleCop Analyzers v1.1.118)_

SA1629 works for the summary of an enum.
However, it ignores the comments on enum members which it should -imho- also check.

Repro:
```C#
    /// <summary>
    /// Some comment.
    /// </summary>
    public enum MySample
    {
        /// <summary>
        /// Missing dot
        /// </summary>
        None = 0,

        /// <summary>
        /// Ends with dot.
        /// </summary>
        Something = 1,
    }
```
**_Expected_**: Missing period is found for documentation of `None` 
**_Actual_**: No SA1629 violation

**Comment by sharwell** [comment]
😕 Looks like everything that extends `ElementDocumentationBase` ignores these. Weird.

**Comment by VitaliAntonov** [comment]
I investigated further and found that the following analyzers derived from `ElementDocumentationBase` have nothing to do with enum members (in my opinion):
 - [SA1611](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1611.md) ElementParametersMustBeDocumented
 - [SA1612](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1612.md) ElementParameterDocumentationMustMatchElementParameters
 - [SA1613](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1613.md) ElementParameterDocumentationMustDeclareParameterName
 - [SA1614](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1614.md) ElementParameterDocumentationMustHaveText
 - [SA1616](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1616.md) ElementReturnValueDocumentationMustHaveText

Also, I assume that the remaining two rules should take into account enum members:
- [SA1608](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1608.md) ElementDocumentationMustNotHaveDefaultSummary
  ```csharp
  public enum EnumName
  {
      /// <summary>
      /// Summary description for EnumMember.
      /// </summary>
      EnumMember = 0,
  }
  ```
- [SA1625](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1625.md) ElementDocumentationMustNotBeCopiedAndPasted
  ```csharp
  public enum EnumName
  {
      /// <summary>
      /// Some documentation.
      /// </summary>
      /// <remark>Some documentation.</remark>
      EnumMember = 0,
  }
  ```

I'll register a handler for enum members in `ElementDocumentationBase` and add unit tests for [SA1629](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1629.md), [SA1608](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1608.md) and [SA1625](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1625.md).
